File: codechef/2021 - August Long/probelm difficulties.py
t = int(input())
res ={1:2, 2:2, 3:1, 4:0}
for y in range(t):
    diff = list(map(int, input().split()))
    dis = len(set(diff))
    if dis == 4 or dis == 3:
        print(2)
    elif dis == 2:
        diff.sort()
        if diff[1] != diff[2]:
            print(2)
        else:
            print(1)
    else:
        print(0)

File: codechef/2021 - August Long/probelm difficulties.cpp
#include <iostream>
using namespace std;

int distinct(int arr[], int len){
    int dis = 0;
    for (int i=0; i<len; i++){
        int j = 0;
        for (; j<i; j++)
            if (arr[i] == arr[j])
                break;
        if (i == j)
            dis += 1;    
    }
    return dis;     
}

int[] sort(int arr[]){
    
}

int main() {
	
    int t;
    cin >> t;
    for (int i=0; i<t; i++){
        int diff[4];
        for (int j=0; j<4; j++)
            cin >> diff[j];
        if (distinct(diff,4) == 1)
            cout << 0;
        else if (distinct(diff,4) == 2){
            if (sort(diff)[1]!=sort(diff)[2])
                cout << 2;
            else
                cout << 1;
        }
        else
            cout << 2;
        cout << endl;
    }
	return 0;
}
